### PR TITLE
Refactor AWS key generators to a separate package

### PR DESCRIPTION
**Pull request by arielshaqed**
Allows using them from other packages.

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]
LGTM

File: pkg/auth/service.go [hunk 1]
```diff
@@ -699,8 +689,8 @@ func (s *DBAuthService) ListPolicies(ctx context.Context, params *model.Paginati
 }
 
 func (s *DBAuthService) CreateCredentials(ctx context.Context, username string) (*model.Credential, error) {
-	accessKeyID := genAccessKeyID()
-	secretAccessKey := genSecretAccessKey()
+	accessKeyID := compat.GenAccessKeyID()
+	secretAccessKey := compat.GenSecretAccessKey()
```
Comment (Jonathan-Rosenberg) [inline pkg/auth/service.go:693]:
```suggestion
	accessKeyID := keys.GenAccessKeyID()
	secretAccessKey := keys.GenSecretAccessKey()
```
?

File: pkg/gateway/operations/base.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ func (o *Operation) EncodeError(w http.ResponseWriter, req *http.Request, e erro
 
 func generateHostID() string {
 	const generatedHostIDLength = 8
-	return auth.HexStringGenerator(generatedHostIDLength)
+	return compat.HexStringGenerator(generatedHostIDLength)
```
Comment (Jonathan-Rosenberg) [inline pkg/gateway/operations/base.go:153]:
```suggestion
	return keys.HexStringGenerator(generatedHostIDLength)
```
?

**Comment by Jonathan-Rosenberg** [comment]
Other than the comments it looks good

File: pkg/gateway/operations/base.go (hunk 2 above)
Comment (nopcoder) [reply pkg/gateway/operations/base.go:153]: good catch - I think less inner-packages less issues

File: pkg/auth/service.go (hunk 1 above)
Comment (arielshaqed) [reply pkg/auth/service.go:693]: Yikes!  Bad rename, sorry (and thanks...)

File: pkg/gateway/operations/base.go (hunk 2 above)
Comment (arielshaqed) [reply pkg/gateway/operations/base.go:153]: Agreed.  But exporting all of `auth` drags in too much.  I just want to generate AWS credentials somewhere else.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**